applications/GeoMechanicsApplication/custom_conditions/T_condition.cpp

```diff
@@ -0,0 +1,173 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//
+
+
+// Application includes
+#include "custom_conditions/T_condition.hpp"
+
+namespace Kratos
+{
+
+template<unsigned int TDim, unsigned int TNumNodes>
+Condition::Pointer TCondition<TDim,TNumNodes>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(new TCondition(NewId, GetGeometry().Create(ThisNodes), pProperties));
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim,TNumNodes>::GetDofList(
+    DofsVectorType& rConditionDofList,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    const GeometryType& rGeom = GetGeometry();
+    const unsigned int nDof = 1;
+    const unsigned int conditionSize = TNumNodes * nDof;
+    if (rConditionDofList.size() != conditionSize)
+        rConditionDofList.resize( conditionSize );
+
+    for (unsigned int i = 0; i < TNumNodes; ++i) {
+        rConditionDofList[i] = rGeom[i].pGetDof(TEMPERATURE);
+    }
+
+    KRATOS_CATCH( "" )
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim,TNumNodes>::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix,
+    VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    unsigned int conditionSize = TNumNodes;
```

Comment:
this can be moved to the hear file when defined by constexpr.

Reply:
Done, but by const replacing of constexpr.